switched to GitHubReleasesInfoProvider

WaveBirdDash opened the pull request:
- switched to `GitHubReleasesInfoProvider` to obtain download URL (also obtains version string as `%version%` which can be used for child recipes)
  - added `%asset_regex%` to ensure only the .pkg is downloaded
- changed `%developer%` in munki recipe from Facebook to osquery
- updated munki recipe `%description%` (pulled from https://osquery.readthedocs.io/)
- linted all modified recipes via `plutil -convert xml1`

Supersedes #13.

Commit:
switched to GitHubReleasesInfoProvider

- switched to `GitHubReleasesInfoProvider` to obtain download URL (also obtains version string as `%version%` which can be used for child recipes)

Commit:
removed unused arguments

Commit:
moved GitHub repo to input variable

jbaker10 commented:
Hey sorry for the delay, I will test this tomorrow and then get it merged! Thanks for the PR!

nstrauss commented:
@jbaker10 Ha. I did this just now without realizing there's already a PR. Some concern with no `asset_regex` key set the wrong platform asset may be returned. My PR also changes the developer from Facebook to osquery. Feel free to update from my  fork. 

https://github.com/nstrauss/jbaker10-recipes/tree/master/osquery

jbaker10 commented:
Just tested and all methods continue to work (new and old). I do like the specific regex @nstrauss included, @mpanighetti would you mind adding that and then I can approve and merge?

WaveBirdDash commented:
Happy to!

WaveBirdDash committed:
added asset_regex

- added asset_regex to ensure only the .pkg is downloaded
- linted plst xml

WaveBirdDash committed:
changed developer to osquery

changed developer in munki recipe from Facebook to osquery
- updated munki recipe description (pulled from https://osquery.readthedocs.io)
- linted plst xml

jbaker10 reviewed (approved):
Thanks for the PR!
